libs/git_wrapper/Git_wrapper.ml

```diff
@@ -121,6 +121,52 @@ let files_from_git_ls ~cwd =
   in
   files |> File.Path.of_strings
 
+let get_git_root_path () =
+  let cmd = Bos.Cmd.(v "git" % "rev-parse" % "--show-toplevel") in
+  let path_r = Bos.OS.Cmd.run_out cmd in
+  let result = Bos.OS.Cmd.out_string ~trim:true path_r in
+  match result with
+  | Ok (path, (_, `Exited 0)) -> path
+  | _ -> raise (Error "Could not get git root from git rev-parse")
+
+let run_with_worktree ~commit f =
+  let cwd = Sys.getcwd () |> Fpath.v |> Fpath.to_dir_path in
+  let git_root = get_git_root_path () |> Fpath.v |> Fpath.to_dir_path in
+  let relative_path =
+    match Fpath.relativize ~root:git_root cwd with
+    | Some p -> p
+    | None -> raise (Error "")
+  in
+  let rand_dir () =
+    let uuid = Uuidm.v `V4 in
+    let dir_name = "semgrep_git_worktree_" ^ Uuidm.to_string uuid in
```

Comment:
why do we need this uuidm thing? Filename.get_temp_dir_name is not enough to get a random directory?